homeassistant/components/logbook/__init__.py

```diff
@@ -518,6 +522,9 @@ def _generate_states_query(session, start_day, end_day, old_state, entity_ids):
             (States.last_updated == States.last_changed)
             & States.entity_id.in_(entity_ids)
         )
+        .outerjoin(
+            StateAttributes, (States.attributes_id == StateAttributes.attributes_id)
+        )
```

Comment:
Does this have any performance impact, or is it negligible since the DB can just fetch the attribute with the wanted index?

Reply:
Its negligible since the DB can just fetch the attribute with the index.  In fact it can be faster since it doesn't have to reach the attributes in the states table in the same disk block most of the time since its joined after filtering in most cases.